pkg/core/blockchain.go

```diff
@@ -2327,80 +2327,45 @@ func (bc *Blockchain) SubscribeForNotifications(ch chan *state.ContainedNotifica
 // reading these events might affect other Blockchain functions. Make sure you're
 // not changing the received execution results, as it may affect the
 // functionality of Blockchain and other subscribers.
-func (bc *Blockchain) SubscribeForExecutions(ch chan *state.AppExecResult) {
+func (bc *Blockchain) SubscribeForExecutions(ch chan<- *state.AppExecResult) {
 	bc.subCh <- ch
 }
 
 // UnsubscribeFromBlocks unsubscribes given channel from new block notifications,
 // you can close it afterwards. Passing non-subscribed channel is a no-op, but
 // the method can read from this channel (discarding any read data).
-func (bc *Blockchain) UnsubscribeFromBlocks(ch chan *block.Block) {
-unsubloop:
-	for {
-		select {
-		case <-ch:
-		case bc.unsubCh <- ch:
-			break unsubloop
-		}
-	}
+func (bc *Blockchain) UnsubscribeFromBlocks(ch chan<- *block.Block) {
+	bc.unsubCh <- ch
```

Comment:
OK, that's the case @roman-khimov told about in https://github.com/nspcc-dev/neo-go/pull/2884#issuecomment-1400233258. We can't remove this `unsubloop` code, because starting from the unsubscription moment all data sent to this channel should be discarded.